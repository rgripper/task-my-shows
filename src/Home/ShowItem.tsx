import * as React from "react";
import styles from "./ShowItem.scss";
import * as classNames from "classnames";

export const ShowItem = (props: {
  posterUrl: string;
  year: number;
  title: string;
  rating: number;
}) => (
  <article className={styles.root}>
    <img src={props.posterUrl} alt="" />
    <header>
      <small>{props.year}</small>
      <div>{props.title}</div>
      <button className={styles["button-icon"]}>
        <i className="fas fa-heart" />
      </button>
    </header>
    <section className={styles['section-play']}>
      <button className={classNames(styles["button-icon"], styles["button-icon-lg"])} title="Open in youtube">
        <i className="fab fa-youtube"></i>
      </button>
    </section>
    <section>
      <small>Rating</small>
      <div>{Math.floor(props.rating * 10)}%</div>
    </section>
    <section>
      <button className={styles["button-icon"]}>
        <i className="fas fa-clipboard-list" />
      </button>
    </section>
  </article>
);
