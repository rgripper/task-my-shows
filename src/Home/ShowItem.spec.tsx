import * as React from "react";
import { shallow } from "enzyme";
import { ShowItem } from "./ShowItem";

describe("<ShowItem />", () => {
  const setup = (onPlayTrailer: () => void) => {
    return shallow(<ShowItem posterUrl="" rating={80} title="Some title" year={2018} onPlayTrailer={onPlayTrailer}/>);
  };

  it("should render successfully", () => {
    const onPlayTrailer = jest.fn();
    const wrapper = setup(onPlayTrailer);

    expect(wrapper).toMatchSnapshot();

  });

  it("should have a clickable open-trailer button", () => {
    const onPlayTrailer = jest.fn();
    const wrapper = setup(onPlayTrailer);

    const icon = wrapper.find('button > .fa-youtube');
    expect(icon.exists());
    icon.parent().simulate('click');
    expect(onPlayTrailer).toBeCalled();
  });
});
