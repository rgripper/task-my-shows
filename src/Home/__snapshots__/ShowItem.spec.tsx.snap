// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowItem /> should render successfully 1`] = `
<article
  className="root"
>
  <img
    alt=""
    src=""
  />
  <header>
    <small>
      2018
    </small>
    <div>
      Some title
    </div>
    <button
      className="button-icon"
    >
      <i
        className="fas fa-heart"
      />
    </button>
  </header>
  <section
    className="section-play"
  >
    <button
      className="button-icon button-icon-lg"
      onClick={[MockFunction]}
      title="Open in youtube"
    >
      <i
        className="fab fa-youtube"
      />
    </button>
  </section>
  <section>
    <small>
      Rating
    </small>
    <div>
      800
      %
    </div>
  </section>
  <section>
    <button
      className="button-icon"
    >
      <i
        className="fas fa-clipboard-list"
      />
    </button>
  </section>
</article>
`;
